moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -274,6 +274,16 @@ bool JogCalcs::cartesianJogCalcs(geometry_msgs::TwistStamped& cmd, JogArmShared&
     Eigen::Vector3d translation_vector(cmd.twist.linear.x, cmd.twist.linear.y, cmd.twist.linear.z);
     Eigen::Vector3d angular_vector(cmd.twist.angular.x, cmd.twist.angular.y, cmd.twist.angular.z);
 
+    // Set uncontrolled dimensions to 0 in command frame
+    mutex.lock();
+    if(!shared_variables.control_dimensions[0]) translation_vector(0) = 0;
+    if(!shared_variables.control_dimensions[1]) translation_vector(1) = 0;
+    if(!shared_variables.control_dimensions[2]) translation_vector(2) = 0;
+    if(!shared_variables.control_dimensions[3]) angular_vector(0) = 0;
+    if(!shared_variables.control_dimensions[4]) angular_vector(1) = 0;
+    if(!shared_variables.control_dimensions[5]) angular_vector(2) = 0;
```

Comment:
Maybe you could reduce lines of code here by something like:    (pardon my MATLAB pseudocode)

translation_vector = find(shared_variables.control_dimensions(0:2));
angular_vector = find(shared_variables.control_dimensions(3:5));